extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -88,6 +111,8 @@ function SidebarWithRef(
   // `mounted` mounts the component. `opened` plays the animation.
   const [mounted, setMounted] = useState(false);
   const [opened, setOpened] = useState(false);
+  const [side, setSide] = useState(null);
```

Comment:
Since we control whether or not to start animations based on whether or not this `side` variable is truthy, then we initialize to `null` since we should not have a truthy value to start.  We will do all calculations for side within `useLayoutEffect`.

Reply:
This also protects in the use-case (however unlikely) where the global `document.dir` that is available here differs from the `sidebarElement.ownerDocument.dir`.  So this way we only look at the latter, and if the former is defined, it will not prevent us from checking the latter.

Reply:
At the very minimum we should definitely be able to init this to `sideProp`:

```
const [] =  useState(sideProp);
```

When `sideProp` is specified, we're in a much better shape since everything is immediately known.

Reply:
We certainly can.  If we do this though, the first `useLayoutEffect` will exit early, and we do not get any error checking on `sideProp`.  (So a typo like `lfet` would not be captured).  Thoughts?